internal/vfs/utilities.go

```diff
@@ -81,11 +81,11 @@ func IsImplicitGlob(lastPathComponent string) bool {
 	return !strings.ContainsAny(lastPathComponent, ".*?")
 }
 
-// Reserved characters, forces escaping of any non-word (or digit), non-whitespace character.
-// It may be inefficient (we could just match (/[-[\]{}()*+?.,\\^$|#\s]/g), but this is future
-// proof.
+// Reserved characters - only escape actual regex metacharacters.
+// Go's regexp doesn't support \x escape sequences for arbitrary characters,
```

Comment:
We shouldn't be using regexp at all; does switching to regexp2 fix this?

Reply:
It shouldn't be, given regexp2 has a mode that emulates ECMAScript regexes?

If we are going to use `regexp` and not `regexp2`, then we can probably just use `regexp.QuoteMeta`.